main.c

```diff
@@ -371,9 +371,50 @@ int check_winning_condition(
     int last_coordinate[2],
     int lives
 ) {
-    // Hint: return the updated lives count
-    return lives;
-}
+    //If the Frogger lands on water, decrease lives.
+    if (board[last_coordinate[X]][last_coordinate[Y]].type == WATER)
+        {
+            lives--;
+            printf ("Frogger has fallen into the water! Remaining lives: %d\n", lives);
+
+             if (lives == 0) //Should lives reach 0, player has lost the game (losing condition).
+            {
+                printf ("\n!! Game Over !!\n");
+            }
+            else //If lives have not reached 0, reset Frogger position and print the game board to start again.
+            {
+                reset_frogger (board, last_coordinate);
+                return lives;
+            }
+        }
+
+    //If Frogger lands on a lillypad, player has won the game (winning condition).
+    if (board[last_coordinate[X]][last_coordinate[Y]].type == LILLYPAD)
+        {
+            printf ("\nWahoo!! You Won!\n");
+        }
+
+    //If Frogger hits a bug, decrease lives.
+    if (board[last_coordinate[X]][last_coordinate[Y]].bug != NULL) //Looping to check for bugs. As long as the next node in the linked list is not null, continue checking.
+    {
+        if ((board[last_coordinate[X]][last_coordinate[Y]].bug->row == last_coordinate[X]) && (board[last_coordinate[X]][last_coordinate[Y]].bug->col == last_coordinate[Y])) //if bug is present on Frogger position, decrease lives.
+        {
+            lives--;
+            printf ("Oh no! Frogger hit a bug! Remaining lives: %d\n", lives);
+
+            if (lives == 0) //Should lives reach 0, player has lost the game (losing condition).
+            {
+                printf ("\n!! Game Over !!\n");
+            }
+            else //If lives have not reached 0, reset Frogger position and print the game board to start again.
+            {
+                reset_frogger (board, last_coordinate);
+                return lives;
+            }
+        }
+        }
```

Comment:
1 curly brace too many
```suggestion
        }
```